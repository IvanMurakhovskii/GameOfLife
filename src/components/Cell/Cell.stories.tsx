import React from "react";
import { withKnobs, boolean } from "@storybook/addon-knobs";
import { action } from "@storybook/addon-actions";
import Cell from ".";


export default {
    title: "Cell",
    decorators: [withKnobs],
};

export const cell: React.FC = () => (
    <Cell
        alive={boolean('alive', false)}
        handleClick={action("click")}
    />
);