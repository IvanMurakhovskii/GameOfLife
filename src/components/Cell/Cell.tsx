import styled from '@emotion/styled';
import React, { FC } from 'react'


export interface CellProps {
    alive: boolean,
    handleClick(): void,
}


const CellStyle = styled.td<CellProps>`
    background: ${(props) => props.alive && '#3700B3'};
    background: $primary-color;
    border: 1px solid #DDD;
    height: 1em;
    width: 1em;
    &:hover {
            background: ${(props) => !props.alive && '#DDD'};
        }
`;

const Cell: FC<CellProps> = (props: CellProps) => {
    return (
        <CellStyle {...props}
            onClick={props.handleClick}>
        </CellStyle>
    );
}

export default Cell;