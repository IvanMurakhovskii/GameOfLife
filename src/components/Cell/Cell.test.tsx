import React from 'react';
import { shallow } from "enzyme";
import Cell, { CellProps } from './Cell';

const onHandleClick = jest.fn();

const props: CellProps = {
    alive: true,
    handleClick: onHandleClick,
}

const item = <Cell {...props} />

describe("TodoListItem", () => {
    const wrapper = shallow(item);

    it("render item with data", () => {
        expect(wrapper).toMatchSnapshot();
    });

    it("simulate click Cell should call method", () => {
        wrapper.simulate('click');
        expect(onHandleClick).toHaveBeenCalled();
    });

});
