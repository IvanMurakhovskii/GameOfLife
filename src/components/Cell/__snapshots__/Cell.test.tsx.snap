// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoListItem render item with data 1`] = `
<Styled(td)
  alive={true}
  handleClick={[MockFunction]}
  onClick={[MockFunction]}
/>
`;
